### PR TITLE
Use debian base in docker

### PR DESCRIPTION
**Pull request by Anishka0107**
Current size : 621mb
Older size : 816mb

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -1,25 +1,28 @@
-FROM ubuntu:xenial
+FROM debian:stretch-slim
 
-RUN apt-get update -y && apt-get upgrade -y && apt-get install -y \
-build-essential \
+RUN apt-get update -y && apt-get upgrade -y && apt-get install -y --no-install-recommends \
```
Comment (jmmut) [inline docker/Dockerfile]: is the `apt-get upgrade` necessary? will it install all the updates in the system?

Comment (Anishka0107) [reply docker/Dockerfile]: I guess removing it would be better, as we don't need to upgrade all the packages...

Comment (cyenyxe) [reply docker/Dockerfile]: In fact it seems to be [discouraged](https://docs.docker.com/engine/userguide/eng-image/dockerfile_best-practices/#run).

File: docker/Dockerfile [hunk 2]
```diff
@@ -1,25 +1,28 @@
...
+make \
 cmake \
 libboost-dev \
 libboost-filesystem-dev \
 libboost-program-options-dev \
 libboost-regex-dev \
 libsqlite3-dev \
 ragel \
-wget
+&& apt-get clean \
```
Comment (cyenyxe) [inline docker/Dockerfile]: Adding a comment like `Clean up to reduce layer size` before this line would be useful to make a clear separation of the actions.

**Pull request merged**